pkg/asset/manifests/ingress.go

```diff
@@ -108,13 +108,24 @@ func (ing *Ingress) generateClusterConfig(config *types.InstallConfig) ([]byte,
 			Name: "cluster",
 			// not namespaced
 		},
-		Spec: configv1.IngressSpec{
-			Domain: fmt.Sprintf("apps.%s", config.ClusterDomain()),
-		},
+		Spec: configv1.IngressSpec{},
 		Status: configv1.IngressStatus{
 			DefaultPlacement: defaultPlacement,
 		},
 	}
+	obj.Spec.Domain = fmt.Sprintf("apps.%s", config.ClusterDomain())
+	if config.AWS != nil && len(config.AWS.LBType) != 0 {
+		obj.Spec = configv1.IngressSpec{
```

Comment:
You have a bug here. This is causing you to lose the `obj.Spec.Domain`:

```shell
# cat c/manifests/cluster-ingress-02-config.yml 
apiVersion: config.openshift.io/v1
kind: Ingress
metadata:
  creationTimestamp: null
  name: cluster
spec:
  domain: ""
  loadbalancer:
    platform:
      aws:
        type: nlb
      type: AWS
status:
  defaultPlacement: Workers

```

Instead, you could just specify `LoadBalancer` without overwriting the whole spec:

```go
		obj.Spec.LoadBalancer = configv1.LoadBalancer{
				Platform: configv1.IngressPlatformSpec{
					AWS: &configv1.AWSIngressSpec{
						Type: config.AWS.LBType,
					},
					Type: configv1.AWSPlatformType,
				},
		}
```